bats/tests/utils/spin.bats

```diff
@@ -0,0 +1,74 @@
+load '../helpers/load'
+
+# Verify that enabling Wasm support will install spin plugins and templates
+
+local_setup() {
+    if is_windows; then
+        if using_windows_exe; then
+            SPIN_DATA_DIR=$LOCALAPPDATA
+        else
+            SPIN_DATA_DIR="${XDG_DATA_HOME:-$HOME/.local/share}"
+        fi
+    elif is_macos; then
+        SPIN_DATA_DIR="${HOME}/Library/Application Support"
+    elif is_linux; then
+        SPIN_DATA_DIR="${XDG_DATA_HOME:-$HOME/.local/share}"
+    else
+        skip "Unknown OS: $OS"
+    fi
+    SPIN_DATA_DIR="${SPIN_DATA_DIR}/spin"
+}
+
+cmd_exe() {
+    "${SYSTEMROOT}/system32/cmd.exe" /c "$@"
+}
+
+dir_exists() {
+    if using_windows_exe; then
+        run --separate-stderr cmd_exe if exist "$(host_path "$1")" echo True
```

Comment:
This technically doesn't test that it's a directory…

Reply:
Yeah, but it is close enough. The function is only called with directories as arguments.